### PR TITLE
Wc emails to free [MAILPOET-2382]

### PR DESCRIPTION
**Pull request by JanJakes**

### CODE REVIEW

File: assets/js/src/newsletters/automatic_emails.jsx [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+import _ from 'underscore';
+import Hooks from 'wp-js-hooks';
+import MailPoet from 'mailpoet';
+import AutomaticEmailEventsList from 'newsletters/types/automatic_emails/events_list.jsx';
+import EventsConditions from 'newsletters/automatic_emails/events_conditions.jsx';
+import AutomaticEmailsBreadcrumb from 'newsletters/types/automatic_emails/breadcrumb.jsx';
+import SendEventConditions from 'newsletters/automatic_emails/send_event_conditions.jsx';
+import Listings from 'newsletters/automatic_emails/listings.jsx';
+
+const emails = window.mailpoet_premium_automatic_emails || [];
```
Comment (pavel-mailpoet) [inline assets/js/src/newsletters/automatic_emails.jsx]: I think we should remove `_premium` form here.

File: assets/js/src/newsletters/automatic_emails.jsx [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+const newslettersContainer = document.getElementById('newsletters_container');
+
+if (newslettersContainer && !_.isEmpty(emails)) {
+  const addEmails = (types, that) => {
+    // remove automatic emails declared in Free version if they are declared in Premium
```
Comment (pavel-mailpoet) [inline assets/js/src/newsletters/automatic_emails.jsx]: This comment suggests we don't need this piece of code.

File: assets/js/src/newsletters/automatic_emails.jsx [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+      ))();
+
+      return updatedEmail;
+    });
+
+    return [...existingTypes, ...newTypes];
+  };
+
+  const addEmailsRoutes = (routes) => {
+    // remove routes declared in Free version if they are declared in Premium
```
Comment (pavel-mailpoet) [inline assets/js/src/newsletters/automatic_emails.jsx]: This comment suggests we don't need this piece of code.

File: lib/AutomaticEmails/WooCommerce/Events/FirstPurchase.php [hunk 4]
```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace MailPoet\AutomaticEmails\WooCommerce\Events;
+
+use MailPoet\AutomaticEmails\WooCommerce\Helper as WCPremiumHelper;
```
Comment (pavel-mailpoet) [inline lib/AutomaticEmails/WooCommerce/Events/FirstPurchase.php]: We should probably rename the namespace. But also, shall we move the content of `MailPoet\AutomaticEmails\WooCommerce\Helper`  into WooCommerce helper?

File: tests/acceptance/CreateWooCommerceNewsletterCest.php [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  function createFirstPurchaseEmail(\AcceptanceTester $I) {
+    $I->wantTo('Create and configure a first purchase automatic email');
+
+    $newsletter_title = 'First Purchase Email Creation';
+    $I->activateWooCommerce();
+
+    $I->login();
+    $I->amOnMailpoetPage('Emails');
+    $I->click('[data-automation-id="new_email"]');
+    $I->dontSeeElement('[data-automation-id="woocommerce_premium_feature_notice"]');
```
Comment (pavel-mailpoet) [inline tests/acceptance/CreateWooCommerceNewsletterCest.php]: Those automation ids should be fixed.

File: tests/integration/AutomaticEmails/WooCommerce/Events/PurchasedInCategoryTest.php [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+use MailPoet\WooCommerce\Helper as WCHelper;
+use MailPoet\WP\Functions as WPFunctions;
+
+class PurchasedInCategoryTest extends \MailPoetTest {
+
+  /** @var WCHelper */
+  private $woocommerce_helper;
+
+  /** @var WCPremiumHelper */
+  private $premium_helper;
```
Comment (pavel-mailpoet) [inline tests/integration/AutomaticEmails/WooCommerce/Events/PurchasedInCategoryTest.php]: This name should probably be renamed in all files.

File: assets/js/src/newsletters/automatic_emails.jsx (hunk 2 above)
Comment (JanJakes) [reply assets/js/src/newsletters/automatic_emails.jsx]: Actually, we kind of do. "Free" defines some automatic emails and this function adds more. The functionality is still there. I'll include it in the ongoing ticket I'm preparing.

File: assets/js/src/newsletters/automatic_emails.jsx (hunk 3 above)
Comment (JanJakes) [reply assets/js/src/newsletters/automatic_emails.jsx]: Same here.

File: lib/AutomaticEmails/WooCommerce/Events/FirstPurchase.php (hunk 4 above)
Comment (JanJakes) [reply lib/AutomaticEmails/WooCommerce/Events/FirstPurchase.php]: Removed.

File: tests/acceptance/CreateWooCommerceNewsletterCest.php (hunk 5 above)
Comment (JanJakes) [reply tests/acceptance/CreateWooCommerceNewsletterCest.php]: It doesn't even exist anymore. Removed.

**Comment by JanJakes** [comment]
@codemonkey-jack This needs thorough testing - that all WooCommerce email features work in the Free plugin now and that it doesn't break with Premium from the same branch.

Let's _not_ include it in this week's release - not only it needs testing but I am trying to figure out how to migrate translation strings from one project to another on Transifex.

**Comment by codemonkey-jack** [comment]
While this ticket is ready code-wise to merge, we’re going to wait to merge the ticket until after the release as Jan is looking into the translation issues (moving the premium translation to free). https://mailpoet.slack.com/archives/CB514FU92/p1570437701004600

**Pull request merged**